File: Review-08-Java-Toys/Example-1/KeyboardInput.java
import java.util.Scanner;

/**
 * A basic keyboard Input Stream example
 */
class KeyboardInput {    
    /**
     * Java main function.
     *
     * @param argv command line arguments. By convention
     *     this is usually "args" in Java.
     */
    public static void main(String[] argv)
    {
        Scanner inputStream          = new Scanner(System.in);

        int     inputInt             = 0;
        double  inputDouble          = 0;
        char    inputChar            = '\0';
        boolean inputBoolean         = false;

        String  inputStringNoSpaces  = null;
        String  inputStringWholeLine = null;

        System.out.print("Enter an Integer: ");
        inputInt = inputStream.nextInt();

        System.out.print("Enter a Double: ");
        inputDouble = inputStream.nextDouble();

        System.out.print("Enter a Character: ");
        inputChar = (inputStream.next()).charAt(0);

        System.out.print("Enter a Boolean: ");
        inputBoolean = inputStream.nextBoolean();

        System.out.print("Enter an String (no spaces): ");
        inputStringNoSpaces = inputStream.next();

        // Same as the C++ getline
        // Strip the trailing newline from the previous read
        inputStream.nextLine();

        System.out.print("Enter a String (with spaces): ");
        inputStringWholeLine = inputStream.nextLine();

        System.out.println(); // cout << "\n";
        System.out.println("You Entered:"); // cout << "You Entered:" << "\n";
        System.out.format("  Item %2d: %d\n",    1, inputInt);
        System.out.format("  Item %2d: %4.2f\n", 2, inputDouble);
        System.out.format("  Item %2d: %s\n",    3, inputChar);
        System.out.format("  Item %2d: %b\n",    4, inputBoolean);
        System.out.format("  Item %2d: %s\n",    5, inputStringNoSpaces);
        System.out.format("  Item %2d: %s\n",    6, inputStringWholeLine);
    }
}

File: Review-07-Comparing-Languages/Example-1-Python/keyboard_input.py
"""
A basic keyboard Input Stream example.
"""


def main():
    """
    Python "main" function.
    """

    input_int = int(input("Enter an Integer: "))
    input_float = float(input("Enter a Double: "))
    input_char = input("Enter a Character: ")[0]
    input_boolean = bool(input("Enter a Boolean: "))
    input_string_no_spaces = input("Enter an String (no spaces): ")
    input_string_whole_line = input("Enter a String (with spaces): ")

    print()
    print("You Entered:")
    print(f"  Item  1: {input_int:}")
    print(f"  Item  2: {input_float:4.2f}")
    print(f"  Item  3: {input_char:}")
    print(f"  Item  4: {input_boolean:}")
    print(f"  Item  5: {input_string_no_spaces:}")
    print(f"  Item  6: {input_string_whole_line:}")


if __name__ == "__main__":
    main()
